transformations/dev-prod-mode/dev-prod-mode.md

```diff
@@ -0,0 +1,285 @@
+
+Now move up to the section **Variables** and prepare the variables for your shared code. 
+Click to **initialize the variables** with empty values.
+<br> `outFile`: `demoFile.txt` 
+<br> `outFileContent`: `Hello world!`
+
+{: .image-popup}
+![Screenshot - Initialize Variables](/transformations/dev-prod-mode/04-initialize-variables.png)
+
+### Set Output Mapping
+Now go to the section **File Output Mapping** and click **New File Output**. And because the output of the 
+transformation will be the file `demoFile.txt`, let’s set it as *Source* and `demoOutput` as *Tags*. This means 
+that the output will be stored in Storage as `demoFile.txt` with the tag `demoOutput`. Click **ADD FILE OUTPUT**.
+
+{: .image-popup}
+![Screenshot - Set Output Mapping](/transformations/dev-prod-mode/05-output-mapping.png)
+
+### Run Transformation
+Let’s run the component. 
+
+{: .image-popup}
+![Screenshot - Run Transformation](/transformations/dev-prod-mode/06-run-component.png)
+
+After the job is finished, go to **Storage -- Files**, where you can see the file `demoFile.txt` generated.
+
+{: .image-popup}
+![Screenshot - Generated File ](/transformations/dev-prod-mode/07-generated-file.png)
+
+Now you have everything ready to start testing the Development/Production Mode.
+
+## Create Development Branch
+Imagine you're tasked with changing the name of the output file and adding a bit of logging to the transformation. 
+You're hesitant to do this on a live and business critical transformation in production. This is where 
+Development/Production Mode comes in, giving you the option to test the changes safely before merging them to 
+production. 
+
+To create a new branch, click on your project’s name at the top of the screen (for the purposes of this tutorial, 
+the production project is called *Dev branches demo*). Then click on the green icon **NEW** displayed next to 
+your project’s name. 
+
+{: .image-popup}
+![Screenshot - Create Development Branch](/transformations/dev-prod-mode/08-create-dev-branch.png)
+
+Name your new development branch, for example, *My dev branch*, and click **CREATE DEVELOPMENT BRANCH**  to open it.
+
+{: .image-popup}
+![Screenshot - Name Development Branch](/transformations/dev-prod-mode/09-name-dev-branch.png)
+
+Your new branch will appear right below the name of your production project.
+
+{: .image-popup}
+![Screenshot - Created Development Branch](/transformations/dev-prod-mode/10-dev-branch-created.png)
+
+Once inside the new development branch, let’s run the same transformation you ran in the production branch. 
+Go to **Transformations**, select the transformation *Testing dev/prod mode* and run it. 
+
+{: .image-popup}
+![Screenshot - Run Transformation in Development Branch](/transformations/dev-prod-mode/11-option-edit-vars.png)
+
+Notice that there is an option to edit the variables in the *Variables* dropdown box before you run 
+the configuration. But we don’t need this now.
```

Comment:
no přesně, to je tu úplně navíc :D

Reply:
asi to demo melo ukazat, co vsechno je tam mozny. Mam to teda vyhodit?

Reply:
jo, dal bych tam akorat zminku, ze to funguje i se shared code a variables

Reply:
Myslis jako ze cela ta featura (development branches) funguje i se shared code a variables?

Reply:
jj